src/ADAL.PCL/DefaultEvent.cs

```diff
@@ -0,0 +1,128 @@
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+
+namespace Microsoft.IdentityModel.Clients.ActiveDirectory
+{
+    internal class DefaultEvent : EventsBase
+    {
+        internal static List<Tuple<string, string>> DefaultEvents = new List<Tuple<string, string>>();
+
+        internal DefaultEvent(string eventName)
+        {
+            SetEvent(TelemetryStrings.EventName,eventName);
+            //Fill in the default parameters
+            ApplicationName = PlatformPlugin.PlatformInformation.GetApplicationName();
+            SetEvent(TelemetryStrings.ApplicationName,ApplicationName);
+
+            ApplicationVersion = PlatformPlugin.PlatformInformation.GetApplicationVersion();
+            SetEvent(TelemetryStrings.ApplicationVersion, ApplicationVersion);
+
+            SdkVersion = AdalIdHelper.GetAdalVersion();
+            SetEvent(TelemetryStrings.SdkVersion, SdkVersion);
+
+            //SdkId = AdalIdHelper.GetAssemblyFileVersion();
+
+            DeviceId = PlatformPlugin.PlatformInformation.GetDeviceModel();
+            SetEvent(TelemetryStrings.DeviceId, DeviceId);
+
+            Tenant = IdTokenClaim.TenantId;
+            SetEvent(TelemetryStrings.Tenant, Tenant);
+
+            Issuer = IdTokenClaim.Issuer;
+            SetEvent(TelemetryStrings.Issuer, Issuer);
+
+            Idp = IdTokenClaim.IdentityProvider;
+            SetEvent(TelemetryStrings.Idp, Idp);
+
+            Upn = IdTokenClaim.UPN;
+            SetEvent(TelemetryStrings.Upn, Upn);
+
+            Email = IdTokenClaim.Email;
+            SetEvent(TelemetryStrings.Email, Email);
+
+            PasswordExpiration = IdTokenClaim.PasswordExpiration;
+            SetEvent(TelemetryStrings.PasswordExpiration, PasswordExpiration);
+
+            PasswordChangeUrl = IdTokenClaim.PasswordChangeUrl;
+            SetEvent(TelemetryStrings.PasswordChangeUrl, PasswordChangeUrl);
+
+            FamilyName = IdTokenClaim.FamilyName;
+            SetEvent(TelemetryStrings.FamilyName,FamilyName);
+        }
+
+        internal override void SetEvent(string eventName, string eventParameter)
+        {
+            DefaultEvents.Add(new Tuple<string, string>(eventName, eventParameter));
+        }
+
+        internal override List<Tuple<string, string>> GetEvents(string requestId)
+        {
```

Comment:
agree

---
In reply to: [73766435](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/494#discussion_r73766435) [](ancestors = 73766435)